scripts/loki_transform.py

```diff
@@ -112,8 +112,18 @@ def cli(debug):
               help='Inline member functions for SCC-class transformations.')
 @click.option('--inline-marked/--no-inline-marked', default=True,
               help='Inline pragma-marked subroutines for SCC-class transformations.')
-@click.option('--resolve-sequence-association/--no-resolve-sequence-association', default=False,
-              help='Replace array arguments passed as scalars with arrays.')
+@click.option('--resolve-sequence-association', default='off',
+              type=click.Choice(
+                ['off', 'all', 'inlined-only']
+              ),
+              help=(
+                'Option to replace array arguments passed as scalars with arrays. ' +
+                'Choice `all` does replacement for all procedures as part of a global sanitisation transform. ' +
+                'Choice `inlined-only` applies replacement only for procedures ' +
+                'that will be inlined as a result of specifying options ' +
+                '`--inline-members` or `--inline-marked`. ' + 
+                'If inlining is deactivated, choice `inlined-only` has no effect.'
+              ))
```

Comment:
I'm against overloading the option in such a fashion. This will break existing API.
Instead, we should keep the original option as is and do one of the two following (or ideally both):
1. Add a new bool option `--resolve-sequence-association-inlined-calls/--no-resolve-sequence-association-inlined-calls`
2. Parametrize the inline transformation via the Scheduler config (which is already possible). This will be the preferred option in the future anyway.